internal/terminal/spinner.go

```diff
@@ -1,7 +1,56 @@
 package terminal
 
+import (
+	"time"
+
+	"github.com/briandowns/spinner"
+)
+
 // set of supported spinners
 var (
 	SpinnerCircles = []string{"㊂", "㊀", "㊁"}
 	SpinnerDots    = []string{".  ", ".. ", "...", "   "}
 )
+
+// SpinnerOptions represents the spinner options
+type SpinnerOptions struct {
+	Icon     []string
+	Duration time.Duration
+}
+
+// Spinner is a spinner
+type Spinner interface {
+	Start()
+	Stop()
+	SetMessage(message string)
+}
+
+type uiSpinner struct {
+	s *spinner.Spinner
+}
+
+func newUISpinner(message string, opts SpinnerOptions) *uiSpinner {
+	icon := opts.Icon
+	if len(icon) == 0 {
+		icon = SpinnerCircles
+	}
+
+	duration := opts.Duration
+	if duration == 0 {
```

Comment:
⚠️  haven't really fact-checked this works with `time.Duration`... will be testing around a bit, just wanted to get eyes on this

Reply:
confirmed we are getting the spinners to display as expected